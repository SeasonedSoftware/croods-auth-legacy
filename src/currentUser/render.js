export default ({ render, renderLoading }) => (currentUser, props) => {
  const { fetchingInfo, info } = props

  if (fetchingInfo && renderLoading) {
    return renderLoading()
  }

  return render(info, props)
}
