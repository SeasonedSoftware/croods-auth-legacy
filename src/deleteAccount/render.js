import React from 'react'
import Destroyed from '../Destroyed'

export default props => (destroy, { destroying, destroyed }) => {
  if (destroyed) {
    return <Destroyed {...props} />
  }
  return props.render({ ...props, destroy, destroying })
}
