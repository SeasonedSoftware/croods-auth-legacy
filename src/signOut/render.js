import React from 'react'
import Destroyed from '../Destroyed'

export default ({ render, ...props }) => (
  destroy,
  { destroying, destroyed },
) => {
  if (destroyed) {
    return <Destroyed {...props} />
  }

  return render({ ...props, destroy, destroying })
}
