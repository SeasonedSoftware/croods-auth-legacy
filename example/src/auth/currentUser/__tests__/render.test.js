import React from 'react'
import renderer from 'react-test-renderer'

import render from '../render'

jest.mock('../../../Loading', () => props => <div>Loading</div>)

describe('when there is no info', () => {
  it('calls render', () => {
    const props = {
      render: jest.fn(),
    }
    const currentUser = undefined
    const infoProps = {}

    renderer.create(render(props)(currentUser, infoProps))
    expect(props.render).toBeCalled()
  })
})

describe('when is fetching data', () => {
  it('calls render', () => {
    const props = {
      render: jest.fn(),
    }
    const currentUser = undefined
    const infoProps = { fetchingInfo: true }
    const tree = renderer.create(render(props)(currentUser, infoProps)).toJSON()

    expect(props.render).not.toBeCalled()
    expect(tree).toMatchSnapshot()
  })
})
