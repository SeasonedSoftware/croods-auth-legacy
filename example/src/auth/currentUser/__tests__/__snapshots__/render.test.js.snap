// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when is fetching data calls render 1`] = `
<div>
  Loading
</div>
`;
